fcrepo-http-api/src/main/java/org/fcrepo/http/api/ContentExposingResource.java

```diff
@@ -497,10 +507,25 @@ protected static void addCacheControlHeaders(final HttpServletResponse servletRe
             return;
         }
 
-        final FedoraResource mutableResource = resource instanceof NonRdfSourceDescription
-                ? ((NonRdfSourceDescription) resource).getDescribedResource() : resource;
-        final EntityTag etag = new EntityTag(mutableResource.getEtagValue());
-        final Date date = mutableResource.getLastModifiedDate();
+        final EntityTag etag;
+        final Date date;
+
+        // See note about this code in the javadoc above.
+        if (resource instanceof NonRdfSourceDescription) {
+            final NonRdfSourceDescription description = (NonRdfSourceDescription)resource;
+            // Use a weak ETag for the LDP-RS
+            etag = new EntityTag(description.getDescribedResource().getEtagValue(), true);
+            date = description.getDescribedResource().getLastModifiedDate();
+        } else if (resource instanceof FedoraBinary) {
+            final NonRdfSource binary = (NonRdfSource)resource;
+            // Use a strong ETag for the LDP-NR
+            etag = new EntityTag(binary.getDescription().getEtagValue());
```

Comment:
Despite the text in the javadoc above, swapping the etags/lastmod is not clear to me. For example, if I perform a `GET` request on the binary, I would expect the Etag to reflect the state of the binary. If the binary's description changes, I do not want to have to re-download the binary on a subsequent `GET` because my Etag is now different.